packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -619,11 +610,31 @@ export async function launch({
     throw decohered;
   }
 
+  async function afterCommit(blockHeight, blockTime) {
+    await Promise.resolve()
+      .then(afterCommitCallback)
+      .then(
+        (afterCommitStats = {}) => {
+          controller.writeSlogObject({
+            type: 'cosmic-swingset-after-commit-stats',
+            blockHeight,
+            blockTime,
+            ...afterCommitStats,
+          });
+        },
+        error => {
+          console.warn('Error during afterCommitCallback', error);
+        },
+      );
+  }
+
   async function blockingSend(action) {
     if (decohered) {
       throw decohered;
     }
 
+    await afterCommitWorkDone;
```

Comment:
This causes the "between block" house keeping that chain-main does to actually be done. Previously there could be an overlap if the node was behind and started on the next block immediately.